ui/lib/kv/addon/components/page/secret/version-history.hbs

```diff
@@ -1,11 +1,102 @@
+            </span>
+          </div>
+          {{! icons }}
+          <div class="align-self-center" data-test-icon-holder={{versionData.version}}>
+            {{#if (loose-equal versionData.version @metadata.currentVersion)}}
+              <div>
+                <span class="has-text-success is-size-8 is-block">
+                  <Icon @name="check-circle-fill" />Current
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.deletion_time}}
+              <div>
+                <span class="has-text-grey is-size-8 is-block">
+                  <Icon @name="x-square-fill" />Deleted
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.destroyed}}
+              <div>
+                <span class="has-text-danger is-size-8 is-block">
+                  <Icon @name="x-square-fill" />Destroyed
+                </span>
+              </div>
+            {{/if}}
+          </div>
+          {{! version created date }}
+          <div class="is-size-8 has-text-weight-semibold has-text-grey align-self-center">
+            <ToolTip @verticalPosition="above" @horizontalPosition="center" as |T|>
+              <T.Trigger data-test-tooltip-trigger tabindex="-1">
+                Created
+                {{date-format versionData.created_time "MMM dd, yyyy hh:mm a"}}
+              </T.Trigger>
+              <T.Content @defaultClass="tool-tip smaller-font">
+                <div class="box" data-test-hover-copy-tooltip-text>
+                  {{versionData.created_time}}
+                </div>
+              </T.Content>
+            </ToolTip>
+          </div>
+        </div>
 
-<hr class="is-marginless has-background-gray-200" />
\ No newline at end of file
+        <div class="level-right">
+          <div class="level-item">
+            <PopupMenu @name="version-{{versionData.version}}">
+              <nav class="menu">
+                <ul class="menu-list">
+                  <li>
+                    <LinkTo @route="secret.details" @model={{@metadata.path}} @query={{hash version=versionData.version}}>
+                      View version
+                      {{versionData.version}}
+                    </LinkTo>
+                  </li>
+                  <li>
+                    <LinkTo
+                      @route="secret.details.edit"
+                      @model={{@metadata.path}}
+                      @query={{hash version=versionData.version}}
+                      data-test-create-new-version-from={{versionData.version}}
+                    >
+                      Create new version from
+                      {{versionData.version}}
+                    </LinkTo>
+                  </li>
+                </ul>
+              </nav>
+            </PopupMenu>
+          </div>
+        </div>
+      </div>
+    </LinkedBlock>
+  {{/each}}
+{{else}}
+  <EmptyState
+    @title="You do not have metadata read permissions"
```

Comment:
nit: is this verbiage carried over from before? It feels kind of like how we talk about metadata/data internally. The more customer-facing version seems something like `You do not have permission to read metadata`

Reply:
We never had a empty state before—this lovely verbiage is alllll me :)  I added it just in case the user somehow navigated to this view without read permissions. I'll amend.